### ISSUE TITLE
Upgrade builders to Fedora 40

### ISSUE THREAD

**Issue by praiskup** [fedora-copr/copr#3270]
It is out now: https://fedorapeople.org/groups/schedule/f-40/f-40-key-tasks.html

**Comment by praiskup** [comment]
F41 out now: https://fedorapeople.org/groups/schedule/f-41/f-41-key-tasks.html

**Comment by hroncok** [comment]
As of today, at least 146 rawhide specfiles cannot be SRPM'ed on Fedora \< 41.

 - 9 use a BuildSystem
 - 137 use `%autosetup -C`

(The groups are distinct.)

**Comment by praiskup** [comment]
Note for myself: `ansible` updated on copr-be-dev, do the same on copr-be.

**Comment by praiskup** [comment]
I was able to generate AWS images, but libvirt images are blocked on: `https://pagure.io/cloud-sig/issue/433`

**Comment by praiskup** [comment]
Nah, spent some time on getting things work (for some reaons, selinux was breaking VM log-in, and sshd server).

Then I realized that we have too old ansible on RHEL 8 hypervisors :-( to handle DNF5 -based fedora:
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Base' object has no attribute 'load_config_from_file'

```

**Comment by praiskup** [comment]
I was able to generate the golden images finally, and tests are passing - except for the problems with pyp2spec #3517.

**Comment by praiskup** [comment]
Discussed on today's meeting... and except for the pyp2spec problems, everything seems fine -> and the consensus in #3517 seems to be we should push the move to F41 forward.  Doing so now.

**Comment by praiskup** [comment]
Done: https://lists.fedorahosted.org/archives/list/copr-devel@lists.fedorahosted.org/thread/LTJXSOYFX56QA37F6JDQPCDM2G3C2FTC/

**Comment by praiskup** [comment]
Uh, one P9 machine is missing.

**Comment by hroncok** [comment]
Thanks.

**Comment by praiskup** [comment]
There's also the known problem with old specfiles using `%patchN` commands.  Users should migrate to `%patch -PN`, `%patch N` or `%autosetup` in the best case.
